src/js/utils/string-utils.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Useful string handling utilities.
+ */
+define(['underscore'], function(_) {
+    'use strict';
+
+    var interpolateEscape, escapeInterpolateOptions, HTML;
+
+    /**
+     * Creates an HTML snippet.
+     *
+     * @param {string} text The HTML for the text.
+     * @constructor
+     */
+    function HTMLSnippet(text) {
+        this.text = text;
+    }
+    HTMLSnippet.prototype.toString = function() {
+        return this.text;
+    };
+
+    HTML = function(text) {
+        return new HTMLSnippet(text);
+    };
+
+    /**
+     * Returns an HTML-escaped text string by substituting values into
+     * a format string. Parameters are specified as tokens wrapped by
+     * curly braces, e.g. 'Hello {name}'. The tokens are replaced
+     * by the attribute of the same name from the specified options.
+     *
+     * Note: values will be rendered using their toString methods and then
+     * HTML-escaped. The only exception is that instances of the class HTML
+     * are rendered without escaping as their contract declares that they are
+     * already valid HTML.
+     *
+     * Example:
+     *   StringUtils.interpolateEscape(
+     *       'You are enrolling in {spanStart}{courseName}{spanEnd}',
+     *       {
+     *           courseName: 'Rock & Roll',
+     *           spanStart: StringUtils.HTML('<span class="course-title">'),
+     *           spanEnd: StringUtils.HTML('</span>')
+     *       }
+     *   );
+     *
+     * returns:
+     *   'You are enrolling in <span class="course-title">Rock &amp; Roll</span>'
+     *
+     * @param {string} string The string to be interpolated.
+     * @param {Object} options An optional set of options to the template.
+     * @returns {HTML} The resulting HTML-escaped string.
+     */
+    interpolateEscape = function(string, options) {
+        var escapedOptions = escapeInterpolateOptions(options),
+            result = _.escape(string).replace(/{\w+}/g,
+                function(parameter) {
+                    return String(escapedOptions[parameter.slice(1,-1)]);
+                });
+        return HTML(result);
+    };
+
+    /**
+     * Returns an object with each of the specified options appropriately
+     * escaped. If an option is an HTMLSnippet then it does not need to be
+     * escaped, but otherwise it will be returned with its value escaped.
+     *
+     * @param {Object} options The options passed to interpolateEscape.
+     * @returns {Object} An object with each option appropriately escaped.
+     */
+    escapeInterpolateOptions = function(options) {
+        var property, value,
+            result = {};
+        for (property in options) {
+            if (hasOwnProperty.call(options, property)) {
```

Comment:
That is so much easier! I didn't know about this function before (and amazingly there are no usages of it in any of the platform's code that I can see).